gbasis/base_four_symm.py

```diff
@@ -475,15 +450,32 @@ def construct_array_mix(self, coord_types, **kwargs):
                 block *= cont_four.norm_cont.reshape(
                     1, 1, 1, 1, 1, 1, *block.shape[6:8], *[1 for _ in block.shape[8:]]
                 )
-
                 # transform
                 if type_one == "spherical":
+                    transform_one = generate_transformation(
```

Comment:
Just to comment for context:
I think the reason these were defined before was to avoid recomputing the `transform_one` and `transform_two` for each iteration of the for loop over `k` and `l`.

I don't think it makes any performance difference (because this is not the bottleneck of the code) and may be it's better to test it here especially if it introduces bugs.

Reply:
Thanks David, you're right and I added these back into the loop-body to avoid recomputing them.  I don't know why QA stopped working but I ran tox locally and no issues with linters or tests/coverage were found.